### PR TITLE
[intel] Add `TargetInfo`

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

**Review by prathams417 — APPROVED** [review]

**Comment by etiotto** [comment]
@prathams417 @whitneywhtsang this PR has no associated issue and no explanation on the reason/rationale these changes are necessary.  Can you retroactively open an issue, explain the rationale for this PR, and  link it here.

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef TRITON_CONVERSION_TRITONGPU_TO_LLVM_TARGETINFOINTEL_H
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.h:1]: Please add the copyright info. We should not have files without the proper copyright.

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
+#include "TargetInfo.h"
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp:1]: Please add the copyright info. We should not have files without the proper copyright.

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  return std::nullopt;
+}
+
+bool TargetInfo::supportMaximumMinimum() const {
+  return computeCapability >= 80;
+}
+Value TargetInfo::ballot(ConversionPatternRewriter &rewriter, Location loc,
+                         Type type, Value cmp) const {
+  Value threadMask = int_val(type.getIntOrFloatBitWidth(), -1);
+  return rewriter.create<NVVM::VoteBallotOp>(loc, type, threadMask, cmp);
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp:54]: I fail to understand why we need this code, given that we aren't generating LLVM + NVVM (this is the Intel directory).

**Comment by whitneywhtsang** [comment]
> @prathams417 @whitneywhtsang this PR has no associated issue and no explanation on the reason/rationale these changes are necessary. Can you retroactively open an issue, explain the rationale for this PR, and link it here.

Done.

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp (hunk 3 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp:54]: Addressed in https://github.com/intel/intel-xpu-backend-for-triton/pull/747

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp (hunk 2 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.cpp:1]: Addressed in https://github.com/intel/intel-xpu-backend-for-triton/pull/747

File: third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.h (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/TargetInfo.h:1]: Addressed in https://github.com/intel/intel-xpu-backend-for-triton/pull/747

**Pull request merged**